azure-iot-hub/tests/test_iothub_amqp_client.py

```diff
@@ -34,58 +23,109 @@
         assert amqp_client_obj.send_all_messages.call_count == 1
 
-    @pytest.mark.it("Send Message To Device system prop")
-    def test_send_message_to_device_sys_props(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
-            fake_hostname, fake_shared_access_key_name, fake_shared_access_key
-        )
-        iothub_amqp_client.send_message_to_device(fake_device_id, fake_message, fake_sys_prop)
+    @pytest.mark.it("Sends messages with system properties using the uamqp SendClient")
+    def test_send_message_to_device_sys_prop(self, client, mock_uamqp_SendClient):
+        client.send_message_to_device(fake_device_id, fake_message, fake_sys_prop)
         amqp_client_obj = mock_uamqp_SendClient.return_value
 
         assert amqp_client_obj.queue_message.call_count == 1
         assert amqp_client_obj.send_all_messages.call_count == 1
 
-    @pytest.mark.it("Raises an Exception if send_all_messages Fails")
-    def test_raise_exception_on_send_fail(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
-            fake_hostname, fake_shared_access_key_name, fake_shared_access_key
-        )
+    @pytest.mark.it("Raises an Exception if send_all_messages fails")
+    def test_raise_exception_on_send_fail(self, client, mocker, mock_uamqp_SendClient):
         amqp_client_obj = mock_uamqp_SendClient.return_value
         mocker.patch.object(
             amqp_client_obj, "send_all_messages", {uamqp.constants.MessageState.SendFailed}
         )
         with pytest.raises(Exception):
-            iothub_amqp_client.send_message_to_device(fake_device_id, fake_message, fake_app_prop)
+            client.send_message_to_device(fake_device_id, fake_message, fake_app_prop)
 
-    @pytest.mark.it("Disconnect a Device")
-    def test_disconnect_sync(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
-            fake_hostname, fake_shared_access_key_name, fake_shared_access_key
-        )
+    @pytest.mark.it("Calls close() on the uamqp SendClient when disconnect_sync() is called")
+    def test_disconnect_sync(self, client, mock_uamqp_SendClient):
         amqp_client_obj = mock_uamqp_SendClient.return_value
-        iothub_amqp_client.disconnect_sync()
+        client.disconnect_sync()
 
         assert amqp_client_obj.close.call_count == 1
+
+
+#############################################
+# IoTHubAmqpClientSharedAccessKeyAuth Tests #
+#############################################
+
+
+class IoTHubAmqpClientSharedAccessKeyAuthTestConfig(object):
+    @pytest.fixture
+    def required_kwargs(self):
+        return {
+            "hostname": fake_hostname,
+            "shared_access_key_name": fake_shared_access_key_name,
+            "shared_access_key": fake_shared_access_key,
+        }
+
+    @pytest.fixture
+    def client(self, required_kwargs):
+        return IoTHubAmqpClientSharedAccessKeyAuth(**required_kwargs)
+
```

Comment:
There are no instantiation tests for either of these clients - no validation of any of the auth that makes the classes unique